api/tests/core/bookings/test_api.py

```diff
@@ -630,6 +633,86 @@ def test_no_booking_if_external_booking_fails(
 
             assert Booking.query.count() == 0
 
+        @override_features(ENABLE_CGR_INTEGRATION=True)
+        def test_token_is_cancelled_when_cgr_provider_timeout(self, requests_mock):
+            # Given
+            beneficiary = users_factories.BeneficiaryGrant18Factory()
+            cgr_provider = get_provider_by_local_class("CGRStocks")
+            venue_provider = providers_factories.VenueProviderFactory(provider=cgr_provider)
+            cinema_provider_pivot = providers_factories.CinemaProviderPivotFactory(
+                venue=venue_provider.venue, provider=cgr_provider, idAtProvider=venue_provider.venueIdAtOfferProvider
+            )
+            providers_factories.CGRCinemaDetailsFactory(
+                cinemaUrl="http://cgr-cinema-0.example.com/web_service", cinemaProviderPivot=cinema_provider_pivot
+            )
+            offer_solo = offers_factories.EventOfferFactory(
+                name="Séance ciné solo",
+                venue=venue_provider.venue,
+                subcategoryId=subcategories.SEANCE_CINE.id,
+                lastProviderId=cinema_provider_pivot.provider.id,
+            )
+            stock_solo = offers_factories.EventStockFactory(offer=offer_solo, idAtProviders="123%12354114%CGR#111")
+            # We try to book an offer through their API, oh no, we've got a timeout !
+            requests_mock.get(
+                "http://cgr-cinema-0.example.com/web_service?wsdl", text=soap_definitions.WEB_SERVICE_DEFINITION
+            )
+            # The "ReservationPassCulture" fails with a ReadTimeout and we send an "AnnulationPassCulture" query with the newly generated token
+            post_adapter = requests_mock.post(
+                "http://cgr-cinema-0.example.com/web_service",
+                [{"exc": requests.exceptions.ReadTimeout}, {"text": cgr_fixtures.cgr_annulation_response_template()}],
+            )
+
+            # Of course, the ReadTimeout should still be reraised so the current transaction is
+            # rollback and the booking not commited into the database
```

Comment:
```suggestion
            # rollbacked and the booking not commited into the database
```
Et en fait, `backrolled`, plutôt. Je crois que c'est un `rollback`, mais `to backroll`